pkgs/development/python-modules/databricks-connect/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, pkgs, buildPythonPackage, fetchPypi, six, py4j }:
+
+buildPythonPackage rec {
+  pname = "databricks-connect";
+  version = "7.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "996a9d0f271f6c7edbd2d85b2efb6ff4e58d15222e80f87ca17fdbf224e17056";
+  };
+
+  sourceRoot = ".";
+
+  propagatedBuildInputs = [ py4j six pkgs.jdk ];
+
+  # User have to run tests command 
+  #   databricks-connect test
+  # manually after install and configuration
+  #   databricks-connect configure
+  doCheck = false;
+
+  preFixup = ''
+      substituteInPlace "$out/bin/find-spark-home" \
+      --replace find_spark_home.py .find_spark_home.py-wrapped
+  '';
+
+  JAVA_HOME = "${pkgs.jdk}/lib/openjdk";
```

Comment:
this will only be available during buildtime, if you need it during runtime please do:
```
  makeWrapperArgs = [ "--set JAVA_HOME ${pkgs.jdk}/lib/openjdk" ];
```

This will wrap the entrypoint (`databricks-connect` in this case) with JAVA_HOME. If you need this to be set while using as a python package, then you need to patch the source

Reply:
Figured out that JAVA_HOME is set by jdk. So I removed env set of JAVA_HOME.